### PR TITLE
Issue fixing and targeting support

### PR DESCRIPTION
**Pull request by jitendra-xoriant**
Added New Features as below
Strategy Targeting Segments GET/POST
Strategy Target Values GET/POST improvement
Strategy Day Parts (Multiple)
Parent GET param support
Amount in Audience Segments fixed as per new Documentation
Strategy Concept Fix

### CODE REVIEW

File: src/main/java/com/mediamath/terminalone/models/StrategyConcept.java [hunk 1]
```diff
@@ -117,14 +116,6 @@ public String getEntityname() {
...
 	}
 
-	public int getAdvertiserId() {
-		return advertiser_id;
-	}
-
-	public void setAdvertiserId(int advertiserId) {
-		this.advertiser_id = advertiserId;
-	}
-
```
Comment (karthik-mediamath) [inline src/main/java/com/mediamath/terminalone/models/StrategyConcept.java]: We no longer need advertiser here?

File: src/main/java/com/mediamath/terminalone/models/helper/StrategyHelper.java [hunk 2]
```diff
@@ -48,100 +49,115 @@ public static Form getForm(Strategy entity) {
...
+			    && entity.getDealIds().isEmpty()
+			    && entity.getStrategyDayParts().isEmpty()) 
+		{
 			if (entity.getBidAggresiveness() > 0f) {
 				strategyForm.param("bid_aggressiveness", String.valueOf(entity.getBidAggresiveness()));
 			}
-
-			strategyForm.param("bid_price_is_media_only", Utility.getOneOrZero(entity.isBidPriceIsMediaOnly()));
-
+
```
Comment (karthik-mediamath) [inline src/main/java/com/mediamath/terminalone/models/helper/StrategyHelper.java]: Extra space to be cleared finally.

File: src/main/java/com/mediamath/terminalone/models/helper/StrategyHelper.java [hunk 3]
```diff
@@ -330,6 +361,19 @@ public static Form getForm(Strategy entity) {
...
+				if (sdp != null) {
+					strategyForm.param("day_parts." + inc + ".start_hour", String.valueOf(sdp.getStartHour()));
+					strategyForm.param("day_parts." + inc + ".end_hour", String.valueOf(sdp.getEndHour()));
+					strategyForm.param("day_parts." + inc + ".days", String.valueOf(sdp.getDays()));
+					strategyForm.param("day_parts." + inc + ".user_time", Utility.getOneOrZero(sdp.isUserTime()));
+				}
+			}
+		}
+
 		return Utility.getFilteredForm(strategyForm, "strategy");
```
Comment (karthik-mediamath) [inline src/main/java/com/mediamath/terminalone/models/helper/StrategyHelper.java]: so now we support targeting_segments, target_values and day_parts. right?

File: src/main/java/com/mediamath/terminalone/service/GetService.java [hunk 4]
```diff
@@ -146,6 +146,14 @@ public StringBuilder get(QueryCriteria query) throws ClientException, ParseExcep
...
 			}
 		}
+		//param parent can be id only
+		if(query.parent > 0){
+			if (!("".equals(path.toString())) && path.indexOf("?") != -1) {
+				path.append("&parent=" + query.parent);
+			} else {
+				path.append("?parent=" + query.parent);
+			}
+		}
```
Comment (karthik-mediamath) [inline src/main/java/com/mediamath/terminalone/service/GetService.java]: this is to support target_values to get the child right?

File: src/main/java/com/mediamath/terminalone/models/AudienceSegment.java [hunk 5]
```diff
@@ -33,8 +35,8 @@
   private int id;
   private String name;
   private int parent_audience_segment_id;
-  private float retail_cpm;
-  private float wholesale_cpm;
+  private List<Currency> retail_cpm = new ArrayList<Currency>();
+  private List<Currency>  wholesale_cpm = new ArrayList<Currency>();;
   private String tag;
```
Comment (jitendra-xoriant) [inline src/main/java/com/mediamath/terminalone/models/AudienceSegment.java]: It was already defined in our system as separate model, and already used in some entities. 
As we have copied most of our models fro python api, initially most calculated fields are declared as floats. As in new system, they are defined as object which consists of Value and Currency_code. So replacing them as change requested by client.

File: src/main/java/com/mediamath/terminalone/models/Strategy.java [hunk 6]
```diff
@@ -33,6 +33,14 @@
 	public enum audSegInc {
 		AND, OR
 	}
+	
+	public enum tgtSegExc {
+		AND, OR
+	}
+
+	public enum tgtSegInc {
```
Comment (jitendra-xoriant) [inline src/main/java/com/mediamath/terminalone/models/Strategy.java]: agree. we can add it as an improvement, as this has to get handled for almost each model.

File: src/main/java/com/mediamath/terminalone/models/StrategyConcept.java (hunk 1 above)
Comment (jitendra-xoriant) [inline src/main/java/com/mediamath/terminalone/models/StrategyConcept.java]: No, no document has advertiser_id field now, so removed

File: src/main/java/com/mediamath/terminalone/models/helper/StrategyHelper.java (hunk 2 above)
Comment (jitendra-xoriant) [inline src/main/java/com/mediamath/terminalone/models/helper/StrategyHelper.java]: ok

File: src/main/java/com/mediamath/terminalone/models/helper/StrategyHelper.java (hunk 3 above)
Comment (jitendra-xoriant) [inline src/main/java/com/mediamath/terminalone/models/helper/StrategyHelper.java]: yes all three

File: src/main/java/com/mediamath/terminalone/service/GetService.java (hunk 4 above)
Comment (jitendra-xoriant) [inline src/main/java/com/mediamath/terminalone/service/GetService.java]: did not get your question, can you be more specific by sharing GET URL. I have written one test case for audience_segment using parent, and it works fine

Comment (karthik-mediamath) [inline src/main/java/com/mediamath/terminalone/service/GetService.java]: When we add this parama, it should return the child entities for this parent.

**Pull request merged**